train/trainer.py

```diff
@@ -250,9 +250,15 @@ def make_main_scheduler(self, optimizer):
                          eta_min=self.train_params.min_lr)
         elif self.train_params.lr_scheduler_type == 'Linear':
             sched = torch.optim.lr_scheduler.LinearLR
+            total_expected_batches = (
+                self.train_params.epochs *
+                self.expected_batches_per_epoch *
+                self.train_params.accumulate_grad_batches -
```

Comment:
accumulate_grad_batches joins multiple batches to act as one. e.g. if you have accumulate_grad_batches=2, and your dataloader creates 6 batches [a b c d e f] in that order, you'll effectively have 3 batches (a,b), (c,d), (e,f). so it actually reduces not increases the number of effective batches you have. also given we have a disparity between dataloader batches and "stepped" batches, maybe use clearer names than "batches" so there's no confusion: e.g. `train_dataloader_nbatches` instead of  `expected_batches_per_epoch`, and `expected_main_scheduler_steps` for what you're computing here. `expected_main_scheduler_steps = ((self.train_params.epochs * self.train_dataloader_nbatches) // self.train_params.accumulate_grad_batches) - self.train_params.lr_warm_steps` 

this could be smaller than the true number by one if it doesn't divide perfectly by accumulate_grad_batches, but given this is just for scheduling the learning rate, i think thats fine